tests/network/vmi_infosource.go

```diff
@@ -133,6 +133,8 @@ var _ = SIGDescribe("Infosource", func() {
 				return dummyInterfaceExists(vmi)
 			}, 120*time.Second, 2*time.Second).Should(Equal(true))
 
+			Expect(vmi.Status.Interfaces[0].IP).To(Equal(""))
```

Comment:
I do not know which interface to expect at index 0. The assertion of the list is without order later on.

Reply:
since the vmi.status is currently ordered by domain.spec order (at least for the ones that are in the domain)
then we expect the default one with the original mac to be without ips, and it is the first one on the spec

https://github.com/kubevirt/kubevirt/pull/7053/files#diff-4e9d82caf4611031f90a726f4f574fbb65be97a978e8efccea2da4cfab4388adR99

I can make it more explicit by finding the one with the name "default"
but then it will be little less strict, as it won't enforce it is the first one
wdyt?

Reply:
> AFAIU vmi.status being ordered by domain.spec is an implementation detail, and not part of the API.
> 

It is how virt-handler `UpdateStatus` is implemented indeed, it iterates the domain.spec

> I'm OK with @oshoval 's reasoning if a new test is added that assures his claim - i.e. asserting vmi interface status first entries come from the domain. Not exactly sure how it would look like though.
> 
> Filtering it by name seems preferable (to me). Correct me if I'm wrong: isn't the dom XML interface alias the name of the interface in the VMI spec ?

Not sure about that, but we are using anyhow the MAC in order to add the name to the guest-agent reported interfaces / to correlate them.

Reply:
> > AFAIU vmi.status being ordered by domain.spec is an implementation detail, and not part of the API.
> 
> It is how virt-handler `UpdateStatus` is implemented indeed, it iterates the domain.spec, hence the order

What I mean by this is you cannot assume this is how it works unless it is specified somewhere - i.e. a test asserting that "domain interfaces are listed first in the interface status list" or something similar.

> 
> > Filtering it by name seems preferable (to me). Correct me if I'm wrong: isn't the dom XML interface alias the name of the interface in the VMI spec ?
> 
> Not sure about that, but we are using anyhow the MAC in order to add the name to the guest-agent reported interfaces / to correlate domain.spec entry with guest-agent entry.

I don't see how this is related to what I asked - but that's not important.

> 
> I will strive to find the interface with the "default" and check it has no ips.

My point above was that you should look for "default" because that is the name of the interface on the VMI spec. 

If my claim is correct, you should look for `domain.devices.Interfaces[0].Name` - not `"default"`. I am not  100% sure, but [close to](https://github.com/kubevirt/kubevirt/blob/1b5960a736a279b74322b8a2a14e73a093ac3f34/pkg/network/domainspec/generators.go#L139).

Reply:
as we spoke offline, both are pretty the same, it is either using the const (which is what we are doing on all the interfaces right now), or use the `vmi.spec.domain.devices.Interfaces[0].Name` which derived from the const

Anyhow i will present soon something bit different, i will just use the original mac to find the interface
it is the more accurate, because we want to check that the interface with the original mac has no ips

Reply:
> both are pretty the same, it is either using the const (which is what we are doing on all the interfaces right now), or use the `vmi.spec.domain.devices.Interfaces[0].Name` which derived from the const
> 
> Anyhow i will present soon something bit different, i will just use the original mac to find the interface it is the more accurate, because we want to check that the interface with the original mac has no ips

I see. Makes sense. Looking forward to see how that looks.

Reply:
https://github.com/kubevirt/kubevirt/compare/d36b12358e2171b98df9314f1366069bb1fdab85..e57f8d56d826eb85c15cacbd3e2e49e44a85dde2